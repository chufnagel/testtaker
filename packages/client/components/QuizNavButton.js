import PropTypes from "prop-types";

const QuizNavButton = ({ disabled, direction }) => (
  <button
    disabled={disabled}
    className={`quizNavButton ${direction}`}
    direction={direction}
    type="button"
  >
    {direction}
  </button>
);

QuizNavButton.propTypes = {
  disabled: PropTypes.bool,
  direction: PropTypes.string,
};

QuizNavButton.defaultProps = {
  disabled: true,
  direction: "Next",
};

export default QuizNavButton;
