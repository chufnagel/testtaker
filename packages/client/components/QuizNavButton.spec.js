/* global beforeEach describe, expect, test */
import { shallow } from "enzyme";
import QuizNavButton from "./QuizNavButton";

// QuizNavButton.defaultProps = {
//   disabled: true,
//   direction: "Next",
// };

describe("QuizNavButton", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<QuizNavButton />);
  });
  test("It should render without crashing", () => {
    expect(wrapper.exists()).toEqual(true);
  });
  test("QuizNavButton matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
