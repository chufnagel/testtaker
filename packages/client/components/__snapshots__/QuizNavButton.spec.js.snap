// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizNavButton QuizNavButton matches snapshot 1`] = `
<button
  className="quizNavButton Next"
  direction="Next"
  disabled={true}
  type="button"
>
  Next
</button>
`;
